documentation/specs/Tooltip.md

```diff
@@ -0,0 +1,248 @@
+# `Tooltip` Component Specification
+
+## Overview
+
+A Tooltip shows contextual help or information about specific components when a user hovers or focuses on them.
+
+### Use Cases
+
+- Provide informative text to the user on keyboard focus or mouse hover
+
+### Features
+
+- Custom placement
+- Ability to disable independent of its trigger
+- Custom trigger mode (hover and focus, or just focus)
+- Ability to be controlled or uncontrolled
+
+### Risks and Challenges
+
+- Ensuring tooltips don't get cut off within hidden containers
+- Keeping the API elegantly simple while needing to be aware of inner element `ref`s and needing to pass `prop`s to inner components
+- Will need to ensure that Easy UI components are forwarding refs in order to handle ref attachment
+
+### Prior Art
+
+- [Paste `<Tooltip />`](https://paste.twilio.design/components/tooltip)
+- [Spectrum `<Tooltip />`](https://react-spectrum.adobe.com/react-spectrum/Tooltip.html)
+
+---
+
+## Design
+
+### API
+
+React Aria exposes a lot of functionality for placement and behavior, but the API for Easy UI's `Tooltip` component will be intentionally limited to reduce inconsistency across the UIs. Instead, the component will hold the specific React Aria configuration optimal for EasyPost. We can expose functionality on a per-needed basis as it arises in the future.
+
+```ts
+// prettier-ignore
+type Placement = 'bottom' | 'bottom left' | 'bottom right' | 'bottom start' | 'bottom end' |
+    'top' | 'top left' | 'top right' | 'top start' | 'top end' |
+    'left' | 'left top' | 'left bottom' | 'start' | 'start top' | 'start bottom' |
+    'right' | 'right top' | 'right bottom' | 'end' | 'end top' | 'end bottom';
```

Comment:
**question**:  im trying to visualize these placements, with respect to `bottom` does this diagram align with whats here?

![Screen Shot 2023-04-13 at 1 08 58 PM](https://user-images.githubusercontent.com/44451175/231871965-a27423a7-940f-4825-9207-5412e2fb1324.png)

Reply:
good question. again it is a copy from the same type that's used on Aria's side: https://github.com/adobe/react-spectrum/blob/main/packages/%40react-types/overlays/src/index.d.ts#L16

`bottom` by itself is center like you drew. the difference between `start`, `end`, `left`, and `right` are which values respect the user's language direction. as an example, if you want it to be anchored to the "start" of an element dependent on LTR or RTL support, you'd use `start`; whereas if you wanted it to always be on the "left", you'd use `left`